src/Bicep.Core/Parsing/ParamsParser.cs

```diff
@@ -2,13 +2,14 @@
 // Licensed under the MIT License.
 
 using System.Collections.Immutable;
+using Bicep.Core.Features;
 using Bicep.Core.Syntax;
 
 namespace Bicep.Core.Parsing
 {
     public class ParamsParser : BaseParser
     {
-        public ParamsParser(string text) : base(text)
+        public ParamsParser(string text, IFeatureProvider? featureProvider) : base(text, featureProvider)
```

Comment:
Since `IFeatureProvider` is only needed by `ParamParser` but not `Parser` we can remove it from the base class and make it non-nullable.